src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -86,7 +86,18 @@ public static class TarUtils
                 Task.Run(async () => {
                     try
                     {
-                        await dedupManifestClient.DownloadToStreamAsync(dedupId, process.StandardInput.BaseStream, proxyUri: null, cancellationToken: ct);
+                        await AsyncHttpRetryHelper.InvokeVoidAsync(
+                            async () =>
+                            {
+                                await dedupManifestClient.DownloadToStreamAsync(dedupId, process.StandardInput.BaseStream, proxyUri: null, cancellationToken: ct);
```

Comment:
I think this might be an issue to reuse the previous stream for our retries. We will need a new stream for every retry, or we need to reset the stream starting position to 0. I think we can punt on this for the existing PR.